presto-docs/src/main/sphinx/include/PrestoThriftService.thrift

```diff
@@ -0,0 +1,305 @@
+ * If row is null then value is ignored.
+ */
+struct PrestoThriftTimestamp {
+  1: optional list<bool> nulls;
+  2: optional list<i64> timestamps;
+}
+
+
+/**
+ * Elements of {@code nulls} array determine if a value for a corresponding row is null.
+ * Each elements of {@code sizes} array contains the length in bytes for the corresponding element.
+ * If row is null then the corresponding element in {@code sizes} is ignored.
+ * {@code bytes} array contains uft8 encoded byte values for string representation of json.
+ * Values for all rows are written to {@code bytes} array one after another.
+ * The total number of bytes must be equal to the sum of all sizes.
+ */
+struct PrestoThriftJson {
+  1: optional list<bool> nulls;
+  2: optional list<i32> sizes;
+  3: optional binary bytes;
+}
+
+
+/**
+ * Elements of {@code nulls} array determine if a value for a corresponding row is null.
+ * Each elements of {@code sizes} array contains the length in bytes for the corresponding element.
+ * If row is null then the corresponding element in {@code sizes} is ignored.
+ * {@code bytes} array contains encoded byte values for HyperLogLog representation as defined in airlift specification.
+ * Values for all rows are written to {@code bytes} array one after another.
+ * The total number of bytes must be equal to the sum of all sizes.
+ */
+struct PrestoThriftHyperLogLog {
+  1: optional list<bool> nulls;
+  2: optional list<i32> sizes;
+  3: optional binary bytes;
+}
+
+
+/**
+ * Elements of {@code nulls} array determine if a value for a corresponding row is null.
+ * Each elements of {@code sizes} array contains the number of elements in the corresponding values array.
+ * If row is null then the corresponding element in {@code sizes} is ignored.
+ * {@code values} is a bigint block containing array elements one after another for all rows.
+ * The total number of elements in bigint block must be equal to the sum of all sizes.
+ */
+struct PrestoThriftBigintArray {
+  1: optional list<bool> nulls;
+  2: optional list<i32> sizes;
+  3: optional PrestoThriftBigint values;
+}
+
+struct PrestoThriftId {
+  1:  binary id;
+}
+
+struct PrestoThriftHostAddress {
+  1:  string host;
+  2:  i32 port;
+}
+
+struct PrestoThriftTableMetadata {
+  1:  PrestoThriftSchemaTableName schemaTableName;
+  2:  list<PrestoThriftColumnMetadata> columns;
+  3: optional string comment;
+}
+
+struct PrestoThriftBlock {
+  1: optional PrestoThriftInteger integerData;
+  2: optional PrestoThriftBigint bigintData;
+  3: optional PrestoThriftDouble doubleData;
+  4: optional PrestoThriftVarchar varcharData;
+  5: optional PrestoThriftBoolean booleanData;
+  6: optional PrestoThriftDate dateData;
+  7: optional PrestoThriftTimestamp timestampData;
+  8: optional PrestoThriftJson jsonData;
+  9: optional PrestoThriftHyperLogLog hyperLogLogData;
+  10: optional PrestoThriftBigintArray bigintArrayData;
+}
+
+
+/**
+ * LOWER UNBOUNDED is specified with an empty value and a ABOVE bound
```

Comment:
Done. BTW, I copied it from `Marker`, so it likely needs to be changed there as well.